### ISSUE TITLE
Question: Can you implement subcomands using with this library?

### ISSUE THREAD

**Issue by jr-tutor** [cofyc/argparse#34]
Hi,

does your library support subcomands such as used in GIT or Argparse?

e.g. 
> git add *
> git --help clone
etc. 

Thank you

**Comment by cofyc** [comment]
no builtin support, but it's easy to implement, e.g.

```
struct cmd_struct {
    const char *cmd;
    int len;
    int (*fn) (int, const char **);
    const char *help;
};

static struct cmd_struct commands[] = { 
    {"sub", 3, sub_cmd, NULL},
    ...
}

argc = argparse_parse(&argparse, argc, argv);
if (argc < 1) {
    // show help
}   

/* Try to run command with args provided. */
struct cmd_struct *cmd = NULL;
for (int i = 0; i < ARRAY_SIZE(commands); i++) {
    if (!strcmp(commands[i].cmd, argv[0])) {
        cmd = &commands[i];
    }   
}  

// run the command
```

**Comment by a-random-lemurian** [comment]
Are there plans to add subcommand support to the library?

**Comment by cofyc** [comment]
no, however you can follow [this example](https://github.com/cofyc/argparse/issues/34#issuecomment-688235399)

**Comment by a-random-lemurian** [comment]
> no, however you can follow [this example](https://github.com/cofyc/argparse/issues/34#issuecomment-688235399)

that example just causes a segfault when I `cmd->fn(argc, argv)`, what should I put in place of `// run the command`?

**Comment by cofyc** [comment]
I also create a full example here: https://github.com/cofyc/argparse/blob/master/tests/subcommands.c

**Comment by cofyc** [comment]
If you need to parse arguments for subcommand, you can init and config `argparse` to parse arguments in the same way as in  the main program.

**Comment by cofyc** [comment]
the example is updated, see the [latest version](https://github.com/cofyc/argparse/blob/master/tests/subcommands.c).

Note that `ARGPARSE_STOP_AT_NON_OPTION` must be passed to `argparse_init` in the main program.
